tests/404_or_greater/test_404_or_greater.ml

```diff
@@ -0,0 +1,4 @@
+open Test_helpers
+
+let tests =
+  compile_compare (fun () -> with_bisect () ^ " -w +A-32-4") "404_or_greater"
```

Comment:
Nit: I think we should name the directory and test set as "`test_ppat_open`" or "`test_open_in_patterns`", or something like that.

- From the `4.04` feature list, it seems like the only new thing to deal with.
- If something else gets added in `4.05` or later, it won't go into `404_or_greater`.